q2_phylogeny/_raxml.py

```diff
@@ -0,0 +1,58 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2018, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+import tempfile
+import subprocess
+
+from random import randint
+
+from q2_types.feature_data import AlignedDNAFASTAFormat
+from q2_types.tree import NewickFormat
+
+
+def run_command(cmd, verbose=True):
+    if verbose:
+        print("Running external command line application. This may print "
+              "messages to stdout and/or stderr.")
+        print("The command being run is below. This command cannot "
+              "be manually re-run as it will depend on temporary files that "
+              "no longer exist.")
+        print("\nCommand:", end=' ')
+        print(" ".join(cmd), end='\n\n')
+    subprocess.run(cmd, check=True)
+
+
+def raxml(alignment: AlignedDNAFASTAFormat,
+          seed: int=None,
+          n_threads: int=1,
+          substitution_model: str='GTRGAMMA') -> NewickFormat:
+    aligned_fp = str(alignment)
+    result = NewickFormat()
+
+    if n_threads == 1:
+        cmd = ['raxmlHPC']
+    else:
+        cmd = ['raxmlHPC-PTHREADS', '-T %s' % n_threads]
+
+    if seed is None:
+        seed = randint(1000, 10000)
+
+    runname = 'q2'
+    with tempfile.TemporaryDirectory() as temp_dir:
+        cmd += ['-m', str(substitution_model),
+                '-p', str(seed),
+                '-s', aligned_fp,
+                '-w', temp_dir,
+                '-n', runname]
+        run_command(cmd)
```

Comment:
I haven't taken a deep dive on the raxml docs - are there any other parameters that might be important to expose to q2-users?

Reply:
There. Are. So. Many. Parameters. That is, RAxML uses most of the upper and lower cases of the alphabet for the various flags. Many of which may not apply for our use cases. I decided to keep some parity with what was available in QIIME 1 and add a few more items. I think I will add the rapid bootstrap option. This way users, if interested, can look at the support values of the splits in their tree. But if there are a few other options we should make available, I'd be happy to consider them.

Reply:
I forgot to ask, how to add a citation only when a particular flag is enabled? For example, if I add the ability to perform rapid bootstrapping, then I would like to have that specific citation added to the provenance only if that flag was enabled. I remember discussing something along these lines at the meeting but I am unable to find an example of this.

Reply:
@mikerobeson, that currently isn't possible, a potential future plan would be to create "hidden actions" which are then invoked in a pipeline based on some condition (like that flag). I wonder if that's the right approach for something like this however.

Reply:
@ebolyen okay. To get around this issue in the short term I can setup a separate function called `raxml-bootstrap`. Then I can simply place all relevant citations and extra parameters there. What do you think? Another option is to hold the bootsrap feature for a later pull request?

Reply:
Either works fine I think, and actually an action like raxml-bootstrap might actually make more sense on its own anyhow :+1:, for example you could indicate that the output includes support values on the tree which would be useful to know.

Reply:
I've added `raxml-rapid-bootstrap` as a separate function. I have a couple more simple tests to work out, but the plugin is functional. I do have one question though: how can I properly test if the number of bootstrap replicates I requested is actually being performed / passed in? 

That is, RAxML will write out a single file in which all the individual trees produced for each of these bootstrap replicates resides. For now, this collection of individual bootstrap trees is not retained in the output of `raxml-rapid-bootstrap`, only the final tree with the bootstrap supports is. I am trying to determine the best way to keep this file of trees, or at least return it for testing purposes. I figure, the simplest test would be to count the number of trees written to this file and make sure that is the same count as the number of bootstrap replicates I've requested.

Any suggestions?